Mathlib/Order/WithBot.lean

```diff
@@ -1230,22 +889,28 @@ instance trichotomous.gt [Preorder α] [IsTrichotomous α (· > ·)] :
 instance IsWellOrder.gt [Preorder α] [IsWellOrder α (· > ·)] : IsWellOrder (WithTop α) (· > ·) where
 
 instance _root_.WithBot.trichotomous.lt [Preorder α] [h : IsTrichotomous α (· < ·)] :
-    IsTrichotomous (WithBot α) (· < ·) :=
-  @WithTop.trichotomous.gt αᵒᵈ _ h
+    IsTrichotomous (WithBot α) (· < ·) where
+  trichotomous x y := by cases x <;> cases y <;> simp [trichotomous]
 
 instance _root_.WithBot.isWellOrder.lt [Preorder α] [IsWellOrder α (· < ·)] :
     IsWellOrder (WithBot α) (· < ·) where
 
 instance _root_.WithBot.trichotomous.gt [Preorder α] [h : IsTrichotomous α (· > ·)] :
-    IsTrichotomous (WithBot α) (· > ·) :=
-  @WithTop.trichotomous.lt αᵒᵈ _ h
+    IsTrichotomous (WithBot α) (· > ·) where
+  trichotomous x y := by cases x <;> cases y <;> simp; simpa using trichotomous_of (· > ·) ..
 
 instance _root_.WithBot.isWellOrder.gt [Preorder α] [h : IsWellOrder α (· > ·)] :
-    IsWellOrder (WithBot α) (· > ·) :=
-  @WithTop.IsWellOrder.lt αᵒᵈ _ h
-
-instance [LT α] [DenselyOrdered α] [NoMaxOrder α] : DenselyOrdered (WithTop α) :=
-  OrderDual.denselyOrdered (WithBot αᵒᵈ)
+    IsWellOrder (WithBot α) (· > ·) where
+  trichotomous x y := by cases x <;> cases y <;> simp; simpa using trichotomous_of (· > ·) ..
+
+instance [LT α] [DenselyOrdered α] [NoMaxOrder α] : DenselyOrdered (WithTop α) where
```

Comment:
Same here: this should either keep the def-eq or reason through an order isomorphism.